### PR TITLE
Mastery Exercise Take Two

### PR DESCRIPTION
**Pull request by ben-d-marks**
All proscribed functionality up and running in this MVP.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/lib
```
Comment (JoeShep) [inline .gitignore]: Are you linting your JS? Using Grunt? If so, and your package.json and Gruntfile are in the lib folder, they aren't being pushed up.

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,435 @@
+/*! normalize.css v4.1.1 | MIT License | github.com/necolas/normalize.css */
```
Comment (JoeShep) [inline css/style.css:1]: I would rather see the SCSS files, and have the CSS file gitignored. Makes it easier to review how you're writing the SASS.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          <li class="nav-item">
+            <a class="nav-link" href="#">Some Cars!<span class="sr-only">(current)</span></a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="#">Some More Cars!</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="#">Email Us!</a>
+          </li>
+        </ul>
```
Comment (JoeShep) [inline index.html]: This is behaving a bit oddly in narrower window views. The li items are stacking on top of each other awkwardly

File: javascripts/quiz.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    tempAddition += inventory[i].description;
+    tempAddition += '</p></div><div class="panel-footer"><p>$ ';
+    tempAddition += inventory[i].price.toFixed(2).replace(/(\d)(?=(\d{3})+\.)/g, '$1,');
+    tempAddition += '</p><p>Inventory # ';
+    tempAddition += inventory[i].inventoryNum;
+    tempAddition += '</p></div></div></div><!-- end of card -->';
+    if ( i > 0 && (i + 1) % 3 === 0) {
+      tempAddition += '</div><!-- end of row -->';
+    }
+  }
```
Comment (JoeShep) [inline javascripts/quiz.js:40]: nice solution to the `row` issue

File: index.html [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+<!-- 5. Put a text input field in the navigation bar. -->
+        <input class="col-md-5" type="text" placeholder="Please update the description..." id="descText">
+      </nav>
+    </header>
+
+    <div class="container" id="insert">
```
Comment (JoeShep) [inline index.html:44]: The layout is a bit claustrophobic with the cards never quite fitting in here all at once. Try giving them a little breathing room at the top and bottom

**Review by JoeShep — COMMENTED** [review]
First thing is minor, but really helps when we review these exercises: Name your branch "quiz". That's the branch name we expect when fetching and checking out. (It also indicates carefully following the instructions in the README)

This works quite well. Nice work. I would like to see you handle an attempt to type into the input before a card is clicked. Currently, it throws an error. See if you can anticipate a user doing unexpected things like that and either make it impossible to use the input before a card is clicked, or give the user feedback when they attempt to type before selecting a card.

Other than that, my comments below are really just nitpicks about what you push up vs gitignore and a couple of layout things. I'll give you a +1 once you knock out those fixes.

You're rocking it, sir. I expect great things come December. Keep up the good work!

File: css/style.css (hunk 2 above)
Comment (ben-d-marks) [reply css/style.css:1]: Sorry.. I gitignored from a release perspective, and not a development perspective. Doi! FIXED.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,5 @@
+/lib/node_modules
+/lib/.sass-cache
+*.css.map
```
Comment (ben-d-marks) [inline .gitignore:3]: Notice i reversed the gitignore. Sorry

File: data/inventory.json [hunk 7]
```diff
@@ -0,0 +1,135 @@
+{
```
Comment (ben-d-marks) [inline data/inventory.json:1]: I doubled the number of cars, then added the id of the odd rows to the links in the nav

File: index.html [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    <meta http-equiv="x-ua-compatible" content="ie=edge">
+    <title>Buy a Car!</title>
+    <!-- Bootstrap CSS -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-alpha.5/css/bootstrap.min.css" integrity="sha384-AysaV+vQoT3kOAXZkl02PThvDr8HYKPZhNT5h/CXfBThSRXQ6jW5DO2ekP5ViFdi" crossorigin="anonymous">
+    <link rel="stylesheet" href="css/style.css">
+  </head>
+  <body>
+    <header class="page-header container-fluid">
+<!-- 4. Put a standard Bootstrap navbar element at the top of your page. -->
+      <nav class="navbar navbar-light bg-faded row">
```
Comment (ben-d-marks) [inline index.html:16]: I can't decide if bootstrap caused the stack or not.... either way I rearranged the nav to look better in narrower screens

File: javascripts/activateEvents.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        for (let j = 0; j < cards.length; j++){
+          CarLot.resetCard(cards[j]);
+        }
+        CarLot.formatCard(cards[i], "#a2d2d2");
+      });
+    }
+
+    document.getElementById("descText").addEventListener("keyup", function(){
+      if (!editTarget){
+        alert("Please choose a car...");
```
Comment (ben-d-marks) [inline javascripts/activateEvents.js:25]: Check out this alert that politely asks for a selection....

File: javascripts/quiz.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  // build one massive string to insert there
+  // Loop over the inventory and populate the page
+  var tempAddition = '';
+  for (let i = 0; i < inventory.length; i++){
+    if (i % 3 === 0){
+      tempAddition += '<div class="row">';
+    } 
+    tempAddition += '<div class="col-xs-4 ' + i + '" id="';
+    tempAddition += inventory[i].inventoryNum + i;
+    tempAddition += '"><div class="car-card col-xs-12"><div class="panel panel-default">';
```
Comment (ben-d-marks) [inline javascripts/quiz.js:17]: I made these cards more comfortably spaced by adding a div in my div... within that card div

File: sass/main.scss [hunk 11]
```diff
@@ -0,0 +1,55 @@
+@import '../../../../template';
+
+
```
Comment (ben-d-marks) [inline sass/main.scss:3]: I think i might be doing bad things here.... let me change this.
added template to scss and this PR

**Review by ben-d-marks — COMMENTED** [review]
Funny story: I named this bm-init because I was thinking I'll init the repo, merge then work in quiz after that". Sorry

**Pull request closed without merging**